platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -445,6 +443,29 @@ - (void)_removePack:(MGLOfflinePack *)pack withCompletionHandler:(MGLOfflinePack
             });
         }
     });
+
+}
+
+- (void)invalidatePack:(MGLOfflinePack *)pack withCompletionHandler:(void (^)(NSError * _Nullable))completion {
+    mbgl::OfflineRegion& region = *pack.mbglOfflineRegion;
+    NSError *error;
+    if (!pack.mbglOfflineRegion) {
+        completion(nil);
+        return;
+    }
+
+    _mbglFileSource->invalidateOfflineRegion(region, [&](std::exception_ptr exception) {
+        if (exception) {
```

Comment:
This can move inside the `if (completion)` below.

Reply:
If I move it inside `if (completion)`, will it only happen if there is a completion handler?

Reply:
Yep, but you're only creating the `error` object, and that only gets used in the completion block. (Same thing in the methods below too.)